datafusion/core/src/physical_plan/aggregates/row_hash.rs

```diff
@@ -178,6 +240,28 @@ pub(crate) struct GroupedHashAggregateStream {
 
     /// Have we seen the end of the input
     input_done: bool,
+
```

Comment:
I wonder if we could encapsulate some of this state into its own structure like

```rust
struct SpillState {

    /// If data has previously been spilled, the locations of the
    /// spill files (in Arrow IPC format)
    spills: Vec<NamedTempFile>,

    /// Sorting expression for spilling batches
    spill_expr: Vec<PhysicalSortExpr>,

    /// Schema for spilling batches
    spill_schema: SchemaRef,

    /// Stream of merged outputs after spilling
    merged_stream: SendableRecordBatchStream,

    /// aggregate_arguments for merging spilled data
    merging_aggregate_arguments: Vec<Vec<Arc<dyn PhysicalExpr>>>,

    /// Force spilling always for debugging
    force_spill: bool,
}
```

Reply:
Done